### PR TITLE
Implement Learn Validation with SDP validation

### PR DESCRIPTION
**Pull request by xichan-ms**
[AB#297512](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/297512)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6671)

### CODE REVIEW

File: src/docfx/build/xref/InternalXrefMapBuilder.cs [hunk 1]
```diff
@@ -115,7 +115,7 @@ private IReadOnlyList<InternalXrefSpec> LoadSchemaDocument(ErrorBuilder errors,
         {
             var schema = _templateEngine.GetSchema(_documentProvider.GetMime(file));
 
-            return _jsonSchemaTransformer.LoadXrefSpecs(errors, schema, file, token);
+            return _jsonSchemaTransformer.LoadXrefSpecs(errors.WithCustomRule(schema), schema, file, token);
```
[the remark this answers is not in the document]
Comment (xichan-ms) [reply src/docfx/build/xref/InternalXrefMapBuilder.cs:118]: OK.

**Review by yufeih — APPROVED** [review]

**Comment by OsmondJiang** [comment]
👍👍👍

**Pull request merged**